funsor/affine.py

```diff
@@ -135,10 +134,12 @@ def extract_affine(fn):
         ``(coefficient, eqn)`` pair in einsum form.
     :rtype: tuple
     """
+    # FIXME: does this make sense?
+    prototype = fn.terms[1].data
```

Comment:
@fritzo I need prototype value for the below `ops.new_zeros`. This line passes all gaussian tests but does not work with general `fn` (of course). Do you have an idea to resolve this issue? It seems to me that `extract_affine` is only used in Gaussian. Should we make it a method of gaussian (to use backend information from `info_vec` or `precision`)?

Reply:
A simple solution is to introduce an arg `backend` here, and use Tensor(...).backend to dispatch.

Reply:
This seems reasonable, and preferable to a global option.